### PR TITLE
Fix `s3n` and `s3a` bug by adding them to CLOUD_PROTOCOLS

### PR DESCRIPTION
**Pull request by MigQ2**
## Description
Defining the filepath of a DataSet (e.g. a CSVDataSet) using `s3a://` fails to parse the path as it removes the bucket name from tthe filepath.

This is caused by the `get_protocol_and_path` function, which is called in the `__init__` of most DataSet implementations

https://github.com/kedro-org/kedro/blob/1c6d108b56e405364be3cf1d9cfca65fe6606096/kedro/io/core.py#L700

Currently, defining the DataSet with the `s3://` path instead works fine

Adding `s3a` to `CLOUD_PROTOCOLS` should fix the bug

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by datajoely** [comment]
It looks like fsspec added the [s3a alias in this version](https://filesystem-spec.readthedocs.io/en/latest/changelog.html?highlight=s3a#id22) so this is a great idea. Thank you @MigQ2 !

File: RELEASE.md [hunk 1]
```diff
@@ -30,6 +30,7 @@
 * Fixed bug causing the `after_dataset_saved` hook only to be called for one output dataset when multiple are saved in a single node and async saving is in use.
 * Log level for "Credentials not found in your Kedro project config" was changed from `WARNING` to `DEBUG`.
 * Added safe extraction of tar files in `micropkg pull` to fix vulnerability caused by [CVE-2007-4559](https://github.com/advisories/GHSA-gw9q-c7gh-j9vm).
+* Fixed bug that didn't allow to read or write datasets with `s3a` or `s3n` filepaths
```
Comment (merelcht) [inline RELEASE.md:33]: We've just released `0.18.5`. Would you mind moving this to `0.18.6`?

**Review by merelcht — APPROVED** [review]
Thanks for this addition @MigQ2 ! 👍  Just left one comment about the release notes.

**Review by SajidAlamQB — APPROVED** [review]
Thank you! 🌟

File: RELEASE.md (hunk 1 above)
Comment (MigQ2) [reply RELEASE.md]: Done @merelcht !
Would appreciate if 0.18.6 is released soon 😊

You merge or I do?

Comment (merelcht) [reply RELEASE.md]: We'll do our best!! I'll merge it now. Thanks again ⭐

**Pull request merged**